apps/speed.c

```diff
@@ -561,52 +615,517 @@ static OPT_PAIR ecdh_choices[] = {
 };
 #endif
 
+#ifndef SIGALRM
+# define COND(d) (count < (d))
+# define COUNT(d) (d)
+#else
+# define COND(c) (run && count<0x7fffffff)
+# define COUNT(d) (count)
+# ifndef _WIN32
+/*    signal(SIGALRM, sig_done); */
+# endif
+#endif                         /* SIGALRM */
```

Comment:
This seems to be a repeat of a similar block that appears in the original code later. Why have it twice?
Also, why is the signal code commented out? I'm trying to figure out how the code works without this?